app/models/katello/repository.rb

```diff
@@ -753,21 +754,11 @@ def index_content(full_index = false)
 
     protected
 
-    def removable_unit_association
-      if yum?
-        self.rpms
-      elsif docker?
-        self.docker_manifests
-      elsif puppet?
-        self.puppet_modules
-      elsif ostree?
-        self.ostree_branches
-      elsif file?
-        self.files
-      elsif deb?
-        self.debs
+    def units_from_type_class(type_class = nil)
```

Comment:
awesome work here :+1:

Reply:
You might rename this 'unity_type_for_removal' or something, since it deals with removal type specifically.   Yes this was named poorly prior :)